rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/InlineItemModificationCreator.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client.nodesource;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.ow2.proactive_grid_cloud_portal.rm.client.PluginDescriptor;
+import org.ow2.proactive_grid_cloud_portal.rm.client.nodesource.window.NodeSourceWindow;
+
+import com.smartgwt.client.widgets.form.fields.FormItem;
+import com.smartgwt.client.widgets.form.fields.RadioGroupItem;
+import com.smartgwt.client.widgets.form.fields.TextAreaItem;
+import com.smartgwt.client.widgets.form.fields.UploadItem;
+
+
+public class InlineItemModificationCreator {
+
+    public static final String EDIT_OR_UPLOAD_FORM_ITEM_SUFFIX = ".modify";
+
+    public static final String EDIT_FORM_ITEM_SUFFIX = ".edit";
+
+    public static final String EDIT_RADIO_OPTION_NAME = "edit";
+
+    public static final String UPLOAD_RADIO_OPTION_NAME = "upload";
+
+    private final NodeSourceWindow nodeSourceWindow;
+
+    public InlineItemModificationCreator(NodeSourceWindow nodeSourceWindow) {
+        this.nodeSourceWindow = nodeSourceWindow;
+    }
+
+    public List<FormItem> getModificationChoiceItemsForNonTextualFields(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+        FormItem chooseCredentialsFormItem;
+        List<FormItem> formItemsReplacingNonTextualFormItems = new LinkedList<FormItem>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)